Tests/LoggingTests/LoggingTest.swift

```diff
@@ -417,4 +417,75 @@ class LoggingTest: XCTestCase {
         XCTAssertLessThan(Logger.Level.warning, Logger.Level.critical)
         XCTAssertLessThan(Logger.Level.error, Logger.Level.critical)
     }
+
+    func testStdioLogHandlerOutputsToStderr() {
+        LoggingSystem.bootstrapInternal { _ in
+            StdioLogHandler(label: "test", stream: .stderr)
+        }
+        let log = Logger(label: "test")
+
+        let testString = "test stdout"
+        let interceptedString = String(data: stderr.intercept { log.critical("\(testString)") },
+                                       encoding: .utf8)?.trimmingCharacters(in: .whitespacesAndNewlines)
+
+        let messageSucceeded = interceptedString?.hasSuffix(testString) ?? false
+
+        XCTAssertTrue(messageSucceeded)
+    }
+
+    func testStdioLogHandlerDefaultsToStdout() {
+        LoggingSystem.bootstrapInternal { _ in
+            StdioLogHandler(label: "test")
+        }
+        let log = Logger(label: "test")
+
+        let testString = "test stdout"
+        let interceptedString = String(data: stdout.intercept { log.critical("\(testString)") },
+                                       encoding: .utf8)?.trimmingCharacters(in: .whitespacesAndNewlines)
+
+        let messageSucceeded = interceptedString?.hasSuffix(testString) ?? false
+
+        XCTAssertTrue(messageSucceeded)
+    }
+}
+
+extension UnsafeMutablePointer where Pointee == FILE {
+
+    /// Intercepts data written to a file descriptor during the `intercepted` closure
+    func intercept(_ intercepted: () -> Void) -> Data {
```

Comment:
@allenhumphreys thanks so much for actually testing this (despite it being a pain). I'd be happy with this but I propose an alternative solution up there, let me know what you think about that.